tests/update_reference_data.sh

```diff
@@ -69,68 +167,147 @@ copyDamarisToReferenceDir () {
+#
+# $1 is name of simulator binary
+# $2 is location of reference solutions in opm-tess
+# $3 is base name of simulation input file
+# $4 is test suite's name of test base run
+updateRestartResults () {
+    local binary="${1}"
+    local dir_name="${2}"
+    local file_name="${3}"
+    local test_name="${4}"
 
-# Read failed tests
-FAILED_TESTS=`cat $BUILD_DIR/Testing/Temporary/LastTestsFailed*.log`
-
-test -z "$FAILED_TESTS" && exit 5
-
-for failed_test in $FAILED_TESTS
-do
-  grep -q -E "compareECLFiles|compareDamarisFiles" <<< $failed_test
-  test $? -ne 0 && continue
-  failed_test=`echo $failed_test | sed -e 's/.*://g' -e 's/\+/./g'`
-  # Extract test properties
-  dir=`dirname "$0"`
-  binary=$(awk -v search="set_tests_properties\\\($failed_test\$" -v prop="SIMULATOR" -f $dir/getprop.awk $BUILD_DIR/CTestTestfile.cmake)
-  dir_name=$(awk -v search="set_tests_properties\\\($failed_test\$" -v prop="DIRNAME" -f $dir/getprop.awk $BUILD_DIR/CTestTestfile.cmake)
-  file_name=$(awk -v search="set_tests_properties\\\($failed_test\$" -v prop="FILENAME" -f $dir/getprop.awk $BUILD_DIR/CTestTestfile.cmake)
-  test_name=$(awk -v search="set_tests_properties\\\($failed_test\$" -v prop="TESTNAME" -f $dir/getprop.awk $BUILD_DIR/CTestTestfile.cmake)
-  echo "$failed_test ${binary} ${dirname} ${file_name} ${test_name}"
-  if grep -q compareECLFiles <<< $failed_test
-  then
-    copyToReferenceDir \
-            $BUILD_DIR/tests/results/$binary+$test_name \
-            $OPM_TESTS_ROOT/$dir_name/opm-simulation-reference/$binary \
-            $file_name \
-            EGRID INIT RFT SMSPEC UNRST UNSMRY
-    test $? -eq 0 && changed_tests="$changed_tests $test_name"
+    local rst_steps
+    local rst_step
+    local result
+    local case_result
 
-    if [ -d $configuration/build-opm-simulators/tests/results/$binary+$test_name/restart ]
-    then
+    rst_steps=$(echo ${BUILD_DIR}/tests/results/${binary}+${test_name}/restart/*RESTART_*.UNRST | \
+                    sed -E -e 's;[^ ]*RESTART_([0-9]+)\.UNRST;\1;g')
 
-      RSTEPS=`ls -1 $BUILD_DIR/tests/results/$binary+$test_name/restart/*.UNRST | sed -e 's/.*RESTART_*//' | sed 's/[.].*//' `
-      result=0
-      for RSTEP in $RSTEPS
-      do
+    result=0
+    for rst_step in ${rst_steps}; do
         copyToReferenceDir \
-            $BUILD_DIR/tests/results/$binary+$test_name/restart/ \
-            $OPM_TESTS_ROOT/$dir_name/opm-simulation-reference/$binary/restart \
-            ${file_name}_RESTART_${RSTEP} \
+            "${BUILD_DIR}/tests/results/${binary}+${test_name}/restart/" \
+            "${OPM_TESTS_ROOT}/${dir_name}/opm-simulation-reference/${binary}/restart" \
+            "${file_name}_RESTART_${rst_step}" \
             EGRID INIT RFT SMSPEC UNRST UNSMRY
-        res=$?
-        test $result -eq 0 || result=$res
-      done
-      test $result -eq 0 && changed_tests="$changed_tests $test_name(restart)"
+
+        case_result=$?
+        if [ ${result} -eq 0 ]; then
```

Comment:
since most code is using bash style, maybe stick to it here and elsewhere?

Reply:
> since most code is using bash style, maybe stick to it here and elsewhere?

Sure.  I've pushed an update to that effect now.